lib/features/email/domain/usecases/get_email_content_interactor.dart

```diff
@@ -10,11 +10,14 @@ import 'package:model/extensions/email_extension.dart';
 import 'package:model/extensions/list_attachment_extension.dart';
 import 'package:tmail_ui_user/features/email/domain/repository/email_repository.dart';
 import 'package:tmail_ui_user/features/email/domain/state/get_email_content_state.dart';
+import 'package:tmail_ui_user/features/email/domain/state/get_opened_email_from_cache_state.dart';
+import 'package:tmail_ui_user/features/offline_mode/manager/opened_email_cache_manager.dart';
 
 class GetEmailContentInteractor {
   final EmailRepository emailRepository;
+  final OpenedEmailCacheManager _openedEmailCacheManager;
```

Comment:
Interactor only knows the repository or other interactor, should not know the datasource of the data layer. So you should create a method in the repository and call it here